### DISCUSSION TITLE
result template custom value from Sharepoint List without managed properties

### DISCUSSION THREAD

**Discussion by j1n6** [microsoft-search/pnp-modern-search#3379] [Q&A]
Objectives:
We would like to customise the search result display against a specific Sharepoint List. The Sharepoint List has multiple custom columns. Each result item should show value from the List column value for that item.  We checked that each column has already been auto-crawled. 

Limitation:
Our current site collection has restricted permission to add any Managed Property. Our team does not have permission to configure it. 

Question:
As we understand that the template system only allow us to use Managed Properties, due to the limitation, we currently cannot find a way to customize the search result display using existing Sharepoint list values. Are there any way that we can display list values in the search result without Managed Property configuration?

Thank you for your help in advance!

**Comment by kasperbolarsen** [comment]
Hi

You should be able to get SOME of the way without getting the Site Collection admin permissions. 
When the crawled properties are created a managed property is created sometimes , see the "Auto-generated managed properties" section in https://learn.microsoft.com/en-us/sharepoint/manage-search-schema
You should be able to find and select those managed properties in "Selected Properties " in the result web part. 
Refiners/filters and sortable properties must be mapped using the Site Collection search schema, so I guess you will have to ask an admin to map the crawled properties to refinabledate/string as required

**Comment by j1n6** [comment]
Thank you for providing the guidance. I notice all the Sharepoint List columns are auto-crawled.
I have had looked at the Selected Properties, the Selected Properties does not have auto-crawled property listed. 
Do you mean I still need to add auto-crawled properties to managed properties?

Thanks.

**Comment by j1n6** [comment]
It turned out that we are not allowed to add managed properties. 
Are there any workaround we can still refer to the List values in the search result and customise the display?

**Comment by kasperbolarsen** [comment]
As search is only using the managed Properties  it seems like you are out of luck. 
Is it impossible to ask your IT to cooperate?

**Comment by j1n6** [comment]
Unfortunately, the global site is too big and span across the cooperation, it will take months to get approvals just to make changes to managed property. It will also slow us down if we need to make further changes.

I seem to found a workaround, it involves applying Javascript in the search result template:
Because our search result only targeting a single Sharepoint List (about 3000 entries), and including pagination, this constraints allows us to just use frontend Javascript to query the List fields value.

I have tested by using vanilla Ajax + search result `data.items`, I am able to use List UUID to query field values through existing Sharepoint List API. 

This probably won't scale for other use cases, but for our current need it could work. We can use existing PnP  Search features, with custom template.
